src/spz/spz/iliasharvester.py

```diff
@@ -4,60 +4,223 @@
-from pysimplesoap.simplexml import SimpleXMLElement
+import csv
+import requests
 from spz import app, db, models
 
 
-def get_sid(client, username, password):
-    response = client.login(
-        client='pilot',
-        username=username,
-        password=password
-    )
-    return response.sid
+# headers that will be used for all Ilias HTTPS requests
+headers = {
+    'Accept-Language': 'en-US,en;q=0.8,de-DE;q=0.5,de;q=0.3',
+    'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:44.0) Gecko/20100101 Firefox/44.0',
+}
 
 
-def get_data(client, sid, ref_id):
-    response = client.getTestResults(
-        sid=sid,
-        ref_id=ref_id,
-        sum_only=True
-    )
-    # ilias will return a xml dom, formatted as string and encapsualted in the main responses dom
-    return SimpleXMLElement(str(response.xml))
+def extract_text(s, token_begin, token_end):
+    """Extract text from string that is located between 2 tokens."""
+    l0 = len(token_begin)
+    idx0 = s.find(token_begin)
+    assert idx0 > -1
 
+    begin = idx0 + l0
+
+    idx1 = s.find(token_end, begin)
+    assert idx1 > -1
+
+    end = idx1
+    return s[begin:end]
+
+
+def download_data():
+    """Download relevant CSV data from Ilias.
+
+    Returns line->byte iterator."""
 
-def download_and_parse_data():
-    # get credentials
     username = app.config['ILIAS_USERNAME']
     password = app.config['ILIAS_PASSWORD']
     ref_id = app.config['ILIAS_REFID']
     url = app.config['ILIAS_URL']
 
-    cl = SoapClient(url)
+    # get inital cookies
+    url0 = '{}login.php'.format(url)
+    r0 = requests.get(
+        url0,
+        headers=headers
+    )
+    assert r0.status_code == 200
+    cookies = r0.cookies
+
+    # login
+    url1 = '{}ilias.php'.format(url)
+    r1 = requests.post(
+        url1,
+        params={
+            'baseClass': 'ilStartUpGUI',
+            'cmd': 'post',
+            'cmdClass': 'ilstartupgui',
+            'cmdNode': 'vr',
+            'lang': 'de',
+            'rtoken': '',
+        },
+        data={
+            'cmd[doStandardAuthentication]': 'Anmelden',
+            'password': password,
+            'username': username,
+        },
+        cookies=cookies,
+        headers=headers
+    )
+    assert r1.status_code == 200
+    cookies = r1.history[0].cookies  # use cookies of first request
+
+    # get bunch of metadata
+    url2 = '{}ilias.php'.format(url)
+    r2 = requests.get(
+        url2,
+        params={
+            'baseClass': 'ilrepositorygui',
+            'cmd': 'outEvaluation',
+            'cmdClass': 'iltestevaluationgui',
+            'cmdNode': 'sh:nb:yh',
+            'ref_id': ref_id,
+        },
+        cookies=cookies,
+        headers=headers
+    )
+    assert r2.status_code == 200
+    text2 = r2.text
+    # these tokens occur multiple times but seem to be unique
+    rtoken = extract_text(text2, "rtoken=", "\"")
+    active_id = extract_text(text2, "active_id=", "&")
```

Comment:
Different `active_id`s appear in response HTML and we should make sure that the right one is extracted: It seems that the action of 'form#ilToolbar' is exactly what we want so maybe we could extract the `active_id` from there.

Reply:
Yup, I agree. The token extraction is hacky and unreliable. I copied it from the last ILIAS harvester version :D However, I'll try to figure out something more reliable.

Reply:
Changed the key extraction part to use an html and url parser.